parsely/parselyandroid/ParselyTracker.java

```diff
@@ -57,80 +58,228 @@
 public class ParselyTracker {
     private static ParselyTracker instance = null;
     private static int DEFAULT_FLUSH_INTERVAL = 60;
+    private static int DEFAULT_ENGAGEMENT_INTERVAL_MILLIS = 10500;
     private static String DEFAULT_URLREF = "parsely_mobile_sdk";
 
-    /*! \brief types of post identifiers
-    *
-    *  Representation of the allowed post identifier types
-    */
-    private enum kIdType{ kUrl, kPostId }
+    protected ArrayList<Map<String, Object>> eventQueue;
 
     private String apikey, rootUrl, storageKey, uuidkey, urlref, adKey;
+    private boolean isDebug, flushTimerRunning;
     private SharedPreferences settings;
     private int queueSizeLimit, storageSizeLimit;
-    public int flushInterval;
-    protected ArrayList<Map<String, Object>> eventQueue;
-    private Map<kIdType, String> idNameMap;
     private Map<String, String> deviceInfo;
     private Context context;
     private Timer timer;
+    private FlushManager flushManager;
+    private EngagementManager engagementManager, videoEngagementManager;
+
+    // Here for backwards compat
+    public int flushInterval;
+
+    protected ParselyTracker(String apikey, int flushInterval, String urlref, Context c){
+        this.context = c.getApplicationContext();
+        this.settings = this.context.getSharedPreferences("parsely-prefs", 0);
+
+        this.apikey = apikey;
+        this.uuidkey = "parsely-uuid";
+        this.adKey = null;
+        // get the adkey straight away on instantiation
+        new GetAdKey(c).execute();
+        this.storageKey = "parsely-events.ser";
+        this.rootUrl = "https://srv.pixel.parsely.com/";
+        this.urlref = urlref;
+        this.queueSizeLimit = 50;
+        this.storageSizeLimit = 100;
+        this.deviceInfo = this.collectDeviceInfo();
+        this.timer = new Timer();
+        this.isDebug = false;
+        this.flushInterval = flushInterval;
+
+        this.eventQueue = new ArrayList<>();
+
+        this.flushManager = new FlushManager(this.timer, flushInterval * 1000);
+
+        if(this.getStoredQueue() != null && this.getStoredQueue().size() > 0){
+            this.setFlushTimer();
+        }
+    }
+
+    public double getEngagementInterval() {
+        return DEFAULT_ENGAGEMENT_INTERVAL_MILLIS;
+    }
+
+    public boolean engagementIsActive() {
```

Comment:
Same response as above that it was necessary to keep parity and show things in the example UI.  In this case parity because flush timer state was already public and shown in the example.